### PR TITLE
feat: method to validate the type definitions

### PR DESCRIPTION
**Pull request by tbwiss**
# Description

> **Note**
> 
> Please provide a description of the work completed in this PR below

Adds a public method to validate the type definitions.

## Complexity

> **Note**
>
> Please provide an estimated complexity of this PR of either Low, Medium or High

Complexity: Low

### CODE REVIEW

**Comment by neo4j-team-graphql** [comment]
Looks like you've updated the documentation!

Check out your changes at https://neo4j-graphql-docs-3341.surge.sh

**Comment by neo4j-team-graphql** [comment]
Made some changes to the Neo4j GraphQL Toolbox? Or made some changes to the core library perhaps?

Check out your changes at https://neo4j-graphql-toolbox-3341.surge.sh

Use for instance a local dbms hosted on [Neo4j Desktop](https://neo4j.com/download/) or a cloud-hosted [Aura Free](https://neo4j.com/cloud/aura-free/) dbms.

**Comment by neo4j-team-graphql** [comment]
# Performance Report

No Performance Changes


<details>
    <summary>Show Full Table</summary>
    
| name | dbHits | old dbHits | time (ms) | old time (ms) | maxRows |
| ---- | ------ | ---------- | --------- | ------------- | ------- |
|  aggregations.TopLevelAggregate | 3403 | 3403 | 32 | 53 | 1134 |
|  aggregations.NestedAggregation | 16553 | 16553 | 92 | 99 | 2174 |
|  aggregations.AggregationWithWhere | 11979 | 11979 | 57 | 63 | 2174 |
|  aggregations.AggregationWhereWithinNestedRelationships | 22116987 | 22116987 | 3153 | 3232 | 2008534 |
|  aggregations.AggregationWhereWithinNestedConnections | 22116987 | 22116987 | 2838 | 2867 | 2008534 |
|  aggregations.NestedCountFromMovieToActors | 9734 | 9734 | 41 | 48 | 2174 |
|  aggregations.NestedCountFromActorsToMovie | 9937 | 9937 | 45 | 49 | 2174 |
|  aggregations.DeeplyNestedCount | 13070329 | 13070329 | 4334 | 4773 | 2008534 |
|  batch-create.BatchCreate | 4200 | 4200 | 201 | 137 | 600 |
|  batch-create.BatchCreateSmall | 77 | 77 | 86 | 64 | 11 |
|  connect.createAndConnect | 14424 | 14424 | 200 | 223 | 3003 |
|  connections.Connection | 14082 | 14082 | 66 | 71 | 2174 |
|  connections.NestedConnection | 45472 | 45470 | 116 | 146 | 4516 |
|  connections.ConnectionWithSort | 3284 | 3284 | 65 | 89 | 1040 |
|  connections.ConnectionWithSortAndCypher | 3284 | 3284 | 167 | 267 | 1040 |
|  create.SimpleMutation | 7 | 7 | 53 | 59 | 1 |
|  cypher-directive.TopLevelCypherDirective | 122100 | 122100 | 659 | 637 | 12241 |
|  cypher-directive.TopLevelCypherDirectiveWithColumnName | 153617 | 153617 | 182 | 190 | 12241 |
|  delete.SimpleDelete | 19401 | 19401 | 234 | 362 | 1040 |
|  delete.NestedDeleteInUpdate | 23202 | 23202 | 187 | 215 | 2040 |
|  2925.SingleRelationshipFilter | 6468 | 6468 | 40 | 43 | 1040 |
|  2925.NestedSingleRelationshipFilter | 22900 | 22900 | 78 | 105 | 2174 |
|  2925.SingleRelationshipRequiredFilter | 3450 | 3450 | 25 | 30 | 1134 |
|  2925.NestedSingleRelationshipRequiredFilter | 9383 | 9383 | 47 | 53 | 1040 |
|  query.SimpleQuery | 3121 | 3121 | 20 | 24 | 1040 |
|  query.SimpleQueryWithRelationship | 16162 | 16162 | 43 | 47 | 2174 |
|  query.QueryWhere | 9722 | 9702 | 34 | 40 | 2169 |
|  query.SimpleQueryWithNestedWhere | 9900 | 9880 | 56 | 57 | 2169 |
|  query.Nested | 10096041 | 10096041 | 9842 | 11289 | 2008534 |
|  query.NestedWithFilter | 10074401 | 10074401 | 9978 | 10137 | 2004000 |
|  query.OrFilterOnRelationships | 41190 | 43670 | 243 | 225 | 1287 |
|  query.OrFilterOnRelationshipsAndNested | 34639 | 37539 | 220 | 283 | 1287 |
|  query.QueryWithNestedIn | 13503 | 14217 | 109 | 64 | 1250 |
|  query.NestedConnectionWhere | 9834 | 9834 | 52 | 74 | 2174 |
|  query.DeeplyNestedConnectionWhere | 9906 | 10038 | 71 | 125 | 2174 |
|  query.DeeplyNestedWithRelationshipFilters | 19086 | 19230 | 127 | 300 | 1580 |
|  query.NestedWithRelationshipSingleFilters | 3732 | 3732 | 175 | 197 | 1134 |
|  query.Fulltext | 80 | 80 | 30 | 34 | 16 |
|  query.FulltextWithNestedQuery | 587 | 587 | 47 | 51 | 84 |
|  sorting.SortMultipleTypes | 3515 | 3515 | 64 | 71 | 1040 |
|  sorting.SortMultipleTypesWithCypherWithCypher | 1414 | 1529 | 144 | 241 | 1040 |
|  sorting.SortOnNestedFields | 14082 | 14082 | 105 | 64 | 2174 |
|  sorting.SortDeeplyNestedFields | 43198 | 43198 | 158 | 107 | 4516 |
|  sorting.SortWithTopLevelCypher | 3121 | 3121 | 105 | 166 | 1040 |
|  unions.SimpleUnionQuery | 321 | 321 | 63 | 67 | 35 |
|  unions.SimpleUnionQueryWithMissingFields | 293 | 293 | 59 | 64 | 35 |
|  unions.NestedUnion | 410637 | 410637 | 290 | 311 | 33033 |
|  unions.NestedUnionWithMissingFields | 384611 | 384611 | 288 | 344 | 33033 |
|  update.NestedUpdate | 16143 | 16143 | 125 | 194 | 2002 |

</details>
        
Old Schema Generation: 37.160s
Schema Generation: 37.575s
Old Subgraph Schema Generation: 48.786s
Subgraph Schema Generation: 48.254s

File: packages/graphql/src/classes/Neo4jGraphQL.ts [hunk 1]
```diff
@@ -208,6 +208,35 @@ class Neo4jGraphQL {
         });
     }
 
+    public neo4jValidateGraphQLDocument(): { isValid: boolean; validationErrors: string[] } {
+        let isValid = false;
```
Comment (MacondoExpress) [inline packages/graphql/src/classes/Neo4jGraphQL.ts:212]: As a minor thing I would avoid to mutate the state of these variables, like:
```ts
} catch (error) {
    if (error instanceof Error) {
        const validationErrors = error.message.split("\n\n");
        return { isValid: false, validationErrors };
    }
}
return { isValid: true, validationErrors: [] };
```

**Review by MacondoExpress — DISMISSED** [review]
LGTM!

File: packages/graphql/src/classes/Neo4jGraphQL.ts (hunk 1 above)
Comment (tbwiss) [reply packages/graphql/src/classes/Neo4jGraphQL.ts:212]: Hmm yes, I see your point. I can make the adjustment.

Comment (tbwiss) [reply packages/graphql/src/classes/Neo4jGraphQL.ts]: https://github.com/neo4j/graphql/pull/3341/commits/61ecefcb30069eecd0bdd0ce100e0cd0dcbee81d

**Pull request merged**